symfony/twig-bundle/5.4/templates/base.html.twig

```diff
@@ -14,6 +14,14 @@
         {% endblock %}
     </head>
     <body>
+        {% for type, messages in app.flashes %}
+            {% for message in messages %}
+                <div class="flash flash-{{ type }}">
+                    {{ message }}
+                </div>
+            {% endfor %}
+        {% endfor %}
```

Comment:
For a long time now I have been using in my base template:

```twig
    {% if app.request.hasPreviousSession %}
        {% for type, messages in app.flashes %}
            {% for message in messages %}
                <div class="flash flash-{{ type }}">
                    {{ message }}
                </div>
            {% endfor %}
        {% endfor %}
    {% endif %}
```
specifically to prevent a session from being created when it does not yet exist.

Reply:
`Request::hasPreviousSession()` _initiates_ the session but I think doesn't start it?

Reply:
It just reports if a session exists or not.

Reply:
@nicolas-grekas, what about the above as the default?

Reply:
> It just reports if a session exists or not.

Right, but it would still create the session object from the session factory on every request. That _may not_ be desired.

Reply:
Look at [`Request::hasPreviousSession()`](https://github.com/symfony/symfony/blob/8cd9e0a6ba17df41a1a87f6c297b8f4515ce137c/src/Symfony/Component/HttpFoundation/Request.php#L712). It calls `getSession()`

Reply:
In the `return` statement, it first calls `$this->hasSession()`:
```php
return $this->hasSession() && $this->cookies->has($this->getSession()->getName());
```
 and my understanding is that, if `false === $this->hasSession()` then the part after the `&&` is not evaluated. Only if `true === $this->hasSession()` does PHP check if the part after `&&` is also true or false.

Reply:
By default (if not passing `true` to the `$skipIfUninitialized`) the `hasSession()` will return `true` if it is not `null` and it will never be null if sessions are enabled in your app.